code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -172,14 +173,15 @@
 /obj/structure/closet/proc/after_open(mob/living/user, force = FALSE)
 	return
 
-/obj/structure/closet/proc/insert(atom/movable/AM)
-	if(contents.len >= storage_capacity)
-		return -1
-	if(insertion_allowed(AM))
-		AM.forceMove(src)
-		return TRUE
-	else
+/obj/structure/closet/proc/insert(atom/movable/inserted)
+	if(length(contents) >= storage_capacity)
+		return LOCKER_FULL
+	if(!insertion_allowed(inserted))
 		return FALSE
+	if(SEND_SIGNAL(src, COMSIG_CLOSET_INSERT, inserted))
```

Comment:
Signals must return flags.
Create an appropriate flag, something like `#define COMPONENT_CLOSET_INSERT_INTERRUPT (1<<0)` and then change this to something like `if(SEND_SIGNAL(src, COMSIG_CLOSET_INSERT, inserted) & COMPONENT_CLOSET_INSERT_INTERRUPT)`
And, of course, make the proc registered to the signal return the flag.

Reply:
Done!